packages/next/src/server/app-render/action-handler.ts

```diff
@@ -799,6 +822,7 @@ export async function handleAction({
       type: 'done',
       result: actionResult,
       formState,
+      temporaryReferences: undefined,
```

Comment:
I can't comment on it directly because it isn't part of the diff but I think you need to forward the temporaryReferences to `generateFlight` above when rendering the action result actually.

I'm not sure we need/nor want to use these references in the the re-render that might happen after the handleAction returns because there isn't really a reason to think that render could have a handle on a referred value. So maybe we don't even need this as part of the interface of an action result and can just pass it to generateFlight

Reply:
I didn't do this in [`2fca6bc` (#66054)](https://github.com/vercel/next.js/pull/66054/commits/2fca6bcd2df4fb9b63f05f7c7f6c30fb3025fb2e). In that iteration, it threw when trying to return the previous state from an action.

I tried to pass temporary references into `generateFlight` in [`c954c71` (#66054)](https://github.com/vercel/next.js/pull/66054/commits/c954c715075310cb1e763643883f56ea1e7a767a) (I think) which basically meant going back to passing `temporaryReferences` to `handleAction` instead of creating them in there (since `generateFlight` is called before `handleAction`). But that produced the same error:
1. `pnpm next dev test/e2e/app-dir/actions/`
2. goto `http://localhost:3000/elements`

https://github.com/vercel/next.js/assets/12292047/ac7e68a8-a501-4ba3-8602-2f0af1f788c4